samples/cpp/qrcode.cpp

```diff
@@ -88,42 +93,89 @@ int liveQRCodeDetect(const string& out_file)
         cout << "Cannot open a camera" << endl;
         return -4;
     }
-
+    cout << "Press 'm' to switch between detectAndDecode and detectAndDecodeMulti" << endl;
     QRCodeDetector qrcode;
     TickMeter total;
     for(;;)
     {
-        Mat frame, src, straight_barcode;
-        string decode_info;
-        vector<Point> transform;
-        cap >> frame;
-        if (frame.empty())
+        for(;;)
         {
-            cout << "End of video stream" << endl;
+            Mat frame, src;
+            Mat straight_barcode;
+            string decode_info;
+            vector<Point> transform;
+            cap >> frame;
+            if (frame.empty())
+            {
+                cout << "End of video stream" << endl;
+                break;
+            }
+            cvtColor(frame, src, COLOR_BGR2GRAY);
+            total.start();
```

Comment:
This code should be wrote once. The same note for lines 128-135.

Switch processing/drawing code only.